app/src/main/java/org/mozilla/fenix/settings/advanced/LocaleManagerExtension.kt

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.advanced
+
+import android.content.Context
+import android.content.res.Resources
+import android.os.Build
+import mozilla.components.support.locale.LocaleManager
+import java.util.Locale
+
+/**
+ * Returns a list of currently supported locales, with the system default set as the first one
+ */
+fun LocaleManager.getSupportedLocales(): List<Locale> {
+    val localeList: MutableList<Locale> = ArrayList()
+    localeList.add(getSystemDefault())
+    localeList.add(Locale("eu"))
+    localeList.add(Locale("ca"))
+    localeList.add(Locale("zh", "CN"))
+    localeList.add(Locale("zh", "TW"))
+    localeList.add(Locale("cs"))
+    localeList.add(Locale("nl"))
+    localeList.add(Locale("en"))
+    localeList.add(Locale("fi"))
+    localeList.add(Locale("fr"))
+    localeList.add(Locale("de"))
+    localeList.add(Locale("hu"))
+    localeList.add(Locale("in"))
+    localeList.add(Locale("it"))
+    localeList.add(Locale("ja"))
+    localeList.add(Locale("ko"))
+    localeList.add(Locale("pl"))
+    localeList.add(Locale("pa", "IN"))
+    localeList.add(Locale("ru"))
+    localeList.add(Locale("sk"))
+    localeList.add(Locale("es"))
+    localeList.add(Locale("es", "AR"))
+    localeList.add(Locale("es", "ES"))
+    localeList.add(Locale("sv", "SE"))
+    localeList.add(Locale("ta"))
+    localeList.add(Locale("tr"))
+    return localeList
+}
```

Comment:
Is there a way to do this w/out hard-coding the list?

For one, this list is already out of date, we just added more locales.

Also, we might skip locales as part of the release builds, and then the UI would claim support for locales we don't ship, right?

Reply:
In Focus and FFTV, we have a Gradle task that auto-generates this list based on the the different variants of `/res/values-{android_locale}/strings.xml` directories. We could do something similar here but ideally we were looking to create a Gradle plugin on Android Components to share this functionality between multiple apps, you can find more info here https://github.com/mozilla-mobile/android-components/issues/3586

Reply:
This also another related https://github.com/mozilla-mobile/android-components/issues/3585 issue, hope it helps!

Reply:
@Pike totally agree with removing the hardcoded list, it's just a shame Android has a method in `AssetManager.java` called `getNonSystemLocales()` that is exactly what we need, but it's in their internal API.
I had no idea we had a similar implementation on Focus and FFTV, so I created my own Gradle task, but I feel it's actually a bit simpler than those (for example, using build config instead of generating a java class). 
@Amejia481 please have a look and tell me your opinion :)

Reply:
I left a note on the a-c issue, I came across https://developer.android.com/reference/android/content/res/Configuration.html#getLocales(). But you'd need to practically test it to see if it does the trick. Also in the case of us restricting resources we ship.

Reply:
But regardless of which Gradle task we will use in the end, I agree, we should move it to AC so all 3 products can use it